x-pack/test/api_integration/apis/asset_manager/tests/assets_hosts.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Client } from '@elastic/elasticsearch';
+import {
+  AssetDocument,
+  ESDocumentWithOperation,
+  HostMetricsDocument,
+  infra,
+  PodMetricsDocument,
+  Serializable,
+  timerange,
+} from '@kbn/apm-synthtrace-client';
+import { getDedotTransform } from '@kbn/apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/get_dedot_transform';
+import { getSerializeTransform } from '@kbn/apm-synthtrace/src/lib/apm/client/apm_synthtrace_es_client/get_serialize_transform';
+import { Asset, AssetWithoutTimestamp } from '@kbn/assetManager-plugin/common/types_api';
+import expect from '@kbn/expect';
+import { pipeline, Readable, Transform } from 'stream';
+import { ASSETS_ENDPOINT } from '../constants';
+import { FtrProviderContext } from '../types';
+
+const HOST_ASSETS_ENDPOINT = `${ASSETS_ENDPOINT}/hosts`;
+
+type InfraDocuments =
+  | HostMetricsDocument
+  | PodMetricsDocument
+  | AssetDocument<'host'>
+  | AssetDocument<'pod'>;
+
+export default function ({ getService }: FtrProviderContext) {
+  const supertest = getService('supertest');
+  const es = getService('es');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  describe('asset management', () => {
+    describe('GET /assets/hosts', () => {
+      before(async () => {
+        synthtraceEsClient.pipeline(getInfraPipeline());
+      });
+
+      it('should return host assets from signals', async () => {
+        const end = new Date(); // now
+        const start = new Date();
+        start.setHours(end.getHours() - 1); // now-1h
+
+        const events = timerange(start.getTime(), end.getTime())
+          .interval('10m')
+          .rate(1)
+          .generator<InfraDocuments>((timestamp) => {
+            const hostA = infra.host('host-a');
+            const podA = hostA.pod('pod-a');
+
+            const hostB = infra.host('host-b');
+            const podB = hostA.pod('pod-b');
+
+            return [
+              hostA.metrics(),
+              hostA.asset(),
+              podA.metrics(),
+              podA.asset(),
+              hostB.metrics(),
+              hostB.asset(),
+              podB.metrics(),
+              podB.asset(),
+            ].map(withTimestamp(timestamp));
+          });
+
+        await synthtraceEsClient.index(events); // Only refreshes APM data streams at the end
+        await refreshInfraDataStreams(es);
+
+        const response = await supertest
+          .get(HOST_ASSETS_ENDPOINT)
```

Comment:
The api will either read from signals or asset data depending on the `lockedSource` right ? Is there a way to test both sources or are we bound to testing the one provided in the kibana config ?

Reply:
Currently it's only configurable by changing the Kibana settings so we'd need to setup a new config file for that (similar to how we test the disabled case for the plugin already). I don't feel that's very ergonomic so one idea I had was to put a parameter in the API to override that setting. 
We can leave that setting undocumented but it'll make it easier to test things without multiple configs and files.